test/functional/tests/misc/test_random_cache_ops.py

```diff
@@ -0,0 +1,501 @@
+#
+# Copyright(c) 2024 Huawei Technologies Co., Ltd.
+# SPDX-License-Identifier: BSD-3-Clause
+#
+
+import random
+import pytest
+
+from api.cas import casadm, casadm_parser
+from api.cas.cache_config import (
+    CacheMode,
+    SeqCutOffParameters,
+    SeqCutOffPolicy,
+    CleaningPolicy,
+    FlushParametersAlru,
+    PromotionParametersNhit,
+    PromotionPolicy,
+    FlushParametersAcp,
+)
+from api.cas.casadm_parser import get_caches, get_inactive_cores
+from api.cas.cli_messages import (
+    start_cache_with_existing_metadata,
+    check_stderr_msg,
+    no_cas_metadata,
+    start_cache_with_existing_id,
+)
+from core.test_run import TestRun
+from storage_devices.disk import DiskType, DiskTypeSet, DiskTypeLowerThan
+from test_tools.fio.fio import Fio
+from test_tools.fio.fio_param import IoEngine, ReadWrite
+from test_utils.output import CmdException
+from test_utils.size import Size, Unit
+from datetime import datetime, timedelta
+
+from test_utils.time import Time
+
+io_engine_list = [
+    IoEngine.sync,
+    IoEngine.libaio,
+    IoEngine.psync,
+    IoEngine.pvsync,
+    IoEngine.posixaio,
+    IoEngine.mmap,
+]
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.nand, DiskType.optane]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_random_operations():
+    """
+    title: Stress test for CAS allowed operations.
+    description: |
+      Check that CAS can handle all allowed operation in random run order.
+    pass_criteria:
+      - No system crash.
+    """
+    test_runtime = timedelta(minutes=30)
+
+    with TestRun.step("Prepare disks"):
+        cache_dev = TestRun.disks["cache"]
+        core_dev = TestRun.disks["core"]
+
+        cache_dev.create_partitions([Size(1, Unit.GibiByte)] * 20)
+        core_dev.create_partitions([Size(2, Unit.GibiByte)] * 20)
+
+    with TestRun.step("Start cache and add core"):
+        main_cache = casadm.start_cache(cache_dev=cache_dev.partitions[0], force=True)
+        main_cache.add_core(core_dev=core_dev.partitions[0])
+
+    with TestRun.step("Run random operations"):
+        used_cache_devices = [cache_dev.partitions[0]]
+        available_cache_devices = cache_dev.partitions[1:]
+        used_core_devices = [core_dev.partitions[0]]
+        available_core_devices = core_dev.partitions[1:]
+        devices = {
+            "used_cache_devices": used_cache_devices,
+            "available_cache_devices": available_cache_devices,
+            "used_core_devices": used_core_devices,
+            "available_core_devices": available_core_devices,
+        }
```

Comment:
I really don't really like the idea of putting the lists in a dict and in each operation function getting them back using the string keys. 
Alternatives:
- make the lists global
- define the operation functions within the test so they can access those lists directly
- make a class with the lists as variables  

Preferably one of the first two.